src/main/java/com/google/sps/servlets/DeleteRecipeServlet.java

```diff
@@ -16,11 +25,36 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    long privateRecipeID = Long.parseLong(request.getParameter("id"));
+    String name = request.getParameter("name");
+    String description = request.getParameter("description");
 
-    // PrivateRecipes are the user uploaded recipes so only PrivateRecipes can be deleted (right now)
-    Key recipeEntitykey = KeyFactory.createKey("PrivateRecipe", id);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(recipeEntitykey);
+
+    /// delete the private recipe
+    Key privateKey = KeyFactory.createKey("PrivateRecipe", privateRecipeID);
+    datastore.delete(privateKey);
+
+    // make filters to find the public recipe
+    Filter nameFilter = new FilterPredicate("name", FilterOperator.EQUAL, name);
+    Filter descriptionFilter = new FilterPredicate("description", FilterOperator.EQUAL, description);
+    // TODO: further filter by author ID
+    Filter composFilter = CompositeFilterOperator.and(nameFilter, descriptionFilter);
+
+    Query query = new Query("PublicRecipe").setFilter(composFilter);
+    PreparedQuery results = datastore.prepare(query);
+    
+    try {
+      Entity publicRecipeEntity = results.asSingleEntity();
+      // publicRecipeEntity will only equal null if there is no entity left after the filters are applied
+      if (publicRecipeEntity != null) {
+        long publicRecipeID = publicRecipeEntity.getKey().getId();
+        Key recipeEntitykey = KeyFactory.createKey("PublicRecipe", publicRecipeID);
```

Comment:
This step seems like it isn't needed. You do `publicRecipeEntity.getKey()` already. Why note have one line of `datastore.delete(publicRecipeEntity.getKey())` ?